### PR TITLE
Implementing unit tests for GetConfig

### PR DESCRIPTION
**Pull request by ricardomaraschini**
This PR adds unit tests for function `GetConfig()`. As I am moving now to work on our StorageInterface and this is kinda done I thought it would be good to have this in.

### CODE REVIEW

File: pkg/storage/azure/azure.go [hunk 1]
```diff
@@ -75,47 +77,47 @@ func (e *errDoesNotExist) Error() string {
 	return e.Err.Error()
 }
 
-func getAzureConfigFromCloudSecret(creds *corev1.Secret) (*Azure, error) {
```
Comment (dmage) [inline pkg/storage/azure/azure.go:78]: What's the point of merging these functions?

Comment (ricardomaraschini) [reply pkg/storage/azure/azure.go:78]: I did not see the point of having one function to populate one struct.

Comment (dmage) [reply pkg/storage/azure/azure.go:78]: To be sure that we don't use a wrong secret.

Comment (dmage) [reply pkg/storage/azure/azure.go:78]: We had a bug about that in the past.

Comment (dmage) [reply pkg/storage/azure/azure.go:78]: But tests should level the problem

**Pull request merged**